ddtrace/profiling/collector/memalloc.py

```diff
@@ -100,48 +103,87 @@ def _get_thread_id_ignore_set(self):
 
     def snapshot(self):
         thread_id_ignore_set = self._get_thread_id_ignore_set()
-        return (
-            tuple(
-                MemoryHeapSampleEvent(
-                    thread_id=thread_id,
-                    thread_name=_threading.get_thread_name(thread_id),
-                    thread_native_id=_threading.get_thread_native_id(thread_id),
-                    frames=stack,
-                    nframes=nframes,
-                    size=size,
-                    sample_size=self.heap_sample_size,
+
+        stacks = [
+            ((stack, nframes, thread_id), size)
+            for (stack, nframes, thread_id), size in _memalloc.heap()
+            if not self.ignore_profiler or thread_id not in thread_id_ignore_set
+        ]
+        if self._export_libdd_enabled:
+            for (frames, nframes, thread_id), size in stacks:
+                ddup.start_sample(nframes)
+                ddup.push_heap(size)
+                ddup.push_threadinfo(
+                    thread_id, _threading.get_thread_native_id(thread_id), _threading.get_thread_name(thread_id)
                 )
-                for (stack, nframes, thread_id), size in _memalloc.heap()
-                if not self.ignore_profiler or thread_id not in thread_id_ignore_set
-            ),
-        )
+                ddup.push_class_name(frames[0][3])
+                for frame in frames:
```

Comment:
can we unpack the `frame` collection to make this a bit more readable?

Reply:
Good call--did the `namedtuple` thing and now these frames look like:
```
  ddup.push_frame(frame.function_name, frame.file_name, 0, frame.lineno)
```